### PR TITLE
333 basic component documentation

### PR DESCRIPTION
**Pull request by absurd-turtle**
I added documentation to every component I felt I knew enough about.
Please add or fix the documentation for the components which are created by you.

There is still an issue with the story of CreatableSelectMenu I don't know how to fix.

## Basics

- [x] Details of what you changed are in commit messages.
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [x] The buildserver is happy.
- [x] The PR is rebased with current master.

## Checklist

- [x] I fully described what my PR does in the documentation
      (not in the PR description)
- [x] I fixed all affected documentation
- [x] I fixed all affected decisions
- [ ] I added unit tests for my code
- [x] I added code comments, logging, and assertions as appropriate
- [x] I translated all strings visible to the user
- [x] I mentioned [every code or binary](/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [x] Code is conforming to [our Architecture](/doc/architecture)
- [x] Code is conforming to [our Guidelines](/doc/guidelines)
      (exceptions are documented)
- [x] Code is consistent to [our Design Decisions](/doc/decisions)

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] Examples are well chosen and understandable

### CODE REVIEW

**Review by badnames — APPROVED** [review]
Very nicely done. Thanks.

File: frontend/src/components/Form/CreatableSelectMenu.stories.tsx
Comment (markus2330) [inline frontend/src/components/Form/CreatableSelectMenu.stories.tsx:1]: In storybook I get:
```
Error: Cannot read properties of null (reading 'control')
  at Controller (/node_modules/.cache/.vite-storybook/deps/react-hook-form.js?v=ad66633e:289:42))
  at renderWithHooks (/node_modules/.cache/.vite-storybook/deps/chunk-DHU45P7P.js?v=ad66633e:11763:26))
  at mountIndeterminateComponent (/node_modules/.cache/.vite-storybook/deps/chunk-DHU45P7P.js?v=ad66633e:14492:21))
  at beginWork (/node_modules/.cache/.vite-storybook/deps/chunk-DHU45P7P.js?v=ad66633e:15447:22))
  at beginWork$1 (/node_modules/.cache/.vite-storybook/deps/chunk-DHU45P7P.js?v=ad66633e:19248:22))
  at performUnitOfWork (/node_modules/.cache/.vite-storybook/deps/chunk-DHU45P7P.js?v=ad66633e:18693:20))
  at workLoopSync (/node_modules/.cache/.vite-storybook/deps/chunk-DHU45P7P.js?v=ad66633e:18632:13))
  at renderRootSync (/node_modules/.cache/.vite-storybook/deps/chunk-DHU45P7P.js?v=ad66633e:18611:15))
  at recoverFromConcurrentError (/node_modules/.cache/.vite-storybook/deps/chunk-DHU45P7P.js?v=ad66633e:18233:28))
```

**Review by markus2330 — CHANGES_REQUESTED** [review]
So good to see more components in storybook!

Unfortunately CI fails and I also get plenty of errors locally:

```
npm run doc


> permaplant@0.1.3 doc
> typedoc src

[info] Loaded plugin typedoc-plugin-markdown
src/components/Modals/SimpleModal.test.tsx:10:10 - error TS2741: Property 'onCancel' is missing in type '{ title: string; body: string; setShow: Mock<any, any, any>; show: true; submitBtnTitle: string; cancelBtnTitle: string; onSubmit: Mock<any, any, any>; }' but required in type 'SimpleModalProps'.

10         <SimpleModal
            ~~~~~~~~~~~

  src/components/Modals/SimpleModal.tsx:15:3
    15   onCancel: () => void;
         ~~~~~~~~
    'onCancel' is declared here.

src/features/auth/api/getAuthInfo.ts:1:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'ConfigDto'.

1 import { ConfigDto } from '@/bindings/definitions';
           ~~~~~~~~~

src/features/map_planning/api/createPlanting.ts:1:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'NewPlantingDto'.

1 import { NewPlantingDto, PlantingDto } from '@/bindings/definitions';
           ~~~~~~~~~~~~~~

src/features/map_planning/api/createPlanting.ts:1:26 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'PlantingDto'.

1 import { NewPlantingDto, PlantingDto } from '@/bindings/definitions';
                           ~~~~~~~~~~~

src/features/map_planning/api/getPlantings.ts:1:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'PlantingDto'.

1 import { PlantingDto } from '@/bindings/definitions';
           ~~~~~~~~~~~

src/features/map_planning/api/movePlanting.ts:1:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'PlantingDto'.

1 import { PlantingDto, UpdatePlantingDto } from '@/bindings/definitions';
           ~~~~~~~~~~~

src/features/map_planning/api/movePlanting.ts:1:23 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'UpdatePlantingDto'.

1 import { PlantingDto, UpdatePlantingDto } from '@/bindings/definitions';
                        ~~~~~~~~~~~~~~~~~

src/features/map_planning/api/transformPlanting.ts:1:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'PlantingDto'.

1 import { PlantingDto, UpdatePlantingDto } from '@/bindings/definitions';
           ~~~~~~~~~~~

src/features/map_planning/api/transformPlanting.ts:1:23 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'UpdatePlantingDto'.

1 import { PlantingDto, UpdatePlantingDto } from '@/bindings/definitions';
                        ~~~~~~~~~~~~~~~~~

src/features/map_planning/layers/plant/actions.ts:9:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'PlantingDto'.

9 import { PlantingDto } from '@/bindings/definitions';
           ~~~~~~~~~~~

src/features/map_planning/routes/Map.tsx:10:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'ConnectToMapQueryParams'.

10 import { ConnectToMapQueryParams } from '@/bindings/definitions';
            ~~~~~~~~~~~~~~~~~~~~~~~

src/features/map_planning/store/MapStore.test.ts:6:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'PlantingDto'.

6 import { PlantingDto } from '@/bindings/definitions';
           ~~~~~~~~~~~

src/features/map_planning/store/MapStoreTypes.ts:1:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'PlantingDto'.

1 import { PlantingDto } from '@/bindings/definitions';
           ~~~~~~~~~~~

src/features/map_planning/store/RemoteActions.ts:9:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'Action'.

9 import { Action as RemoteAction } from '@/bindings/definitions';
           ~~~~~~

src/features/map_planning/store/TrackedMapStore.ts:9:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'PlantingDto'.

9 import { PlantingDto } from '@/bindings/definitions';
           ~~~~~~~~~~~

src/features/maps/api/createMap.ts:1:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'MapDto'.

1 import { MapDto, NewMapDto } from '@/bindings/definitions';
           ~~~~~~

src/features/maps/api/createMap.ts:1:18 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'NewMapDto'.

1 import { MapDto, NewMapDto } from '@/bindings/definitions';
                   ~~~~~~~~~

src/features/maps/api/findAllMaps.ts:1:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'MapDto'.

1 import { MapDto, Page } from '@/bindings/definitions';
           ~~~~~~

src/features/maps/components/MapCard.tsx:2:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'MapDto'.

2 import { MapDto } from '@/bindings/definitions';
           ~~~~~~

src/features/maps/components/MapCreateModal.tsx:1:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'NewMapDto'.

1 import { NewMapDto } from '@/bindings/definitions';
           ~~~~~~~~~

src/features/maps/routes/MapOverview.tsx:5:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'NewMapDto'.

5 import { NewMapDto } from '@/bindings/definitions';
           ~~~~~~~~~

src/features/seeds/components/CreateSeedForm.tsx:89:15 - error TS2322: Type '(search: string, options: SelectOption, additional: PageAdditionalInfo | undefined) => Promise<Page>' is not assignable to type 'LoadOptions<SelectOption, GroupBase<SelectOption>, PageAdditionalInfo>'.
  Types of parameters 'options' and 'options' are incompatible.
    Type 'readonly (SelectOption | GroupBase<SelectOption>)[]' is missing the following properties from type 'SelectOption': value, label

89               loadOptions={loadPlants}
                 ~~~~~~~~~~~

  src/components/Form/PaginatedSelectMenu.tsx:45:3
    45   loadOptions: LoadOptions<Option, GroupBase<Option>, PageAdditionalInfo>;
         ~~~~~~~~~~~
    The expected type comes from property 'loadOptions' which is declared here on type 'IntrinsicAttributes & PaginatedSelectMenuProps<NewSeedDto, SelectOption, false>'

src/features/seeds/routes/CreateSeed.tsx:75:10 - error TS2741: Property 'onCancel' is missing in type '{ title: string; body: string; show: boolean; setShow: (showErrorModal: boolean) => void; submitBtnTitle: string; onSubmit: () => void; }' but required in type 'SimpleModalProps'.

75         <SimpleModal
            ~~~~~~~~~~~

  src/components/Modals/SimpleModal.tsx:15:3
    15   onCancel: () => void;
         ~~~~~~~~
    'onCancel' is declared here.

src/features/seeds/routes/SeedDetails.tsx:91:10 - error TS2741: Property 'onCancel' is missing in type '{ title: string; body: string; show: boolean; setShow: Dispatch<SetStateAction<boolean>>; submitBtnTitle: string; onSubmit: () => void; }' but required in type 'SimpleModalProps'.

91         <SimpleModal
            ~~~~~~~~~~~

  src/components/Modals/SimpleModal.tsx:15:3
    15   onCancel: () => void;
         ~~~~~~~~
    'onCancel' is declared here.

src/features/seeds/routes/ViewSeeds.tsx:68:10 - error TS2741: Property 'onCancel' is missing in type '{ title: string; body: string; show: boolean; setShow: Dispatch<SetStateAction<boolean>>; submitBtnTitle: string; onSubmit: () => void; }' but required in type 'SimpleModalProps'.

68         <SimpleModal
            ~~~~~~~~~~~

  src/components/Modals/SimpleModal.tsx:15:3
    15   onCancel: () => void;
         ~~~~~~~~
    'onCancel' is declared here.

src/utils/enum.ts:1:10 - error TS2614: Module '"@/components/Form/SelectMenu"' has no exported member 'SelectOption'. Did you mean to use 'import SelectOption from "@/components/Form/SelectMenu"' instead?

1 import { SelectOption } from '@/components/Form/SelectMenu';
           ~~~~~~~~~~~~


```

**Comment by absurd-turtle** [comment]
> So good to see more components in storybook!
> 
> Unfortunately CI fails and I also get plenty of errors locally:
> 
> ```
> npm run doc
> 
> 
> > permaplant@0.1.3 doc
> > typedoc src
> 
> [info] Loaded plugin typedoc-plugin-markdown
> src/components/Modals/SimpleModal.test.tsx:10:10 - error TS2741: Property 'onCancel' is missing in type '{ title: string; body: string; setShow: Mock<any, any, any>; show: true; submitBtnTitle: string; cancelBtnTitle: string; onSubmit: Mock<any, any, any>; }' but required in type 'SimpleModalProps'.
> 
> 10         <SimpleModal
>             ~~~~~~~~~~~
> 
>   src/components/Modals/SimpleModal.tsx:15:3
>     15   onCancel: () => void;
>          ~~~~~~~~
>     'onCancel' is declared here.
> 
> src/features/auth/api/getAuthInfo.ts:1:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'ConfigDto'.
> 
> 1 import { ConfigDto } from '@/bindings/definitions';
>            ~~~~~~~~~
> 
> src/features/map_planning/api/createPlanting.ts:1:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'NewPlantingDto'.
> 
> 1 import { NewPlantingDto, PlantingDto } from '@/bindings/definitions';
>            ~~~~~~~~~~~~~~
> 
> src/features/map_planning/api/createPlanting.ts:1:26 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'PlantingDto'.
> 
> 1 import { NewPlantingDto, PlantingDto } from '@/bindings/definitions';
>                            ~~~~~~~~~~~
> 
> src/features/map_planning/api/getPlantings.ts:1:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'PlantingDto'.
> 
> 1 import { PlantingDto } from '@/bindings/definitions';
>            ~~~~~~~~~~~
> 
> src/features/map_planning/api/movePlanting.ts:1:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'PlantingDto'.
> 
> 1 import { PlantingDto, UpdatePlantingDto } from '@/bindings/definitions';
>            ~~~~~~~~~~~
> 
> src/features/map_planning/api/movePlanting.ts:1:23 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'UpdatePlantingDto'.
> 
> 1 import { PlantingDto, UpdatePlantingDto } from '@/bindings/definitions';
>                         ~~~~~~~~~~~~~~~~~
> 
> src/features/map_planning/api/transformPlanting.ts:1:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'PlantingDto'.
> 
> 1 import { PlantingDto, UpdatePlantingDto } from '@/bindings/definitions';
>            ~~~~~~~~~~~
> 
> src/features/map_planning/api/transformPlanting.ts:1:23 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'UpdatePlantingDto'.
> 
> 1 import { PlantingDto, UpdatePlantingDto } from '@/bindings/definitions';
>                         ~~~~~~~~~~~~~~~~~
> 
> src/features/map_planning/layers/plant/actions.ts:9:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'PlantingDto'.
> 
> 9 import { PlantingDto } from '@/bindings/definitions';
>            ~~~~~~~~~~~
> 
> src/features/map_planning/routes/Map.tsx:10:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'ConnectToMapQueryParams'.
> 
> 10 import { ConnectToMapQueryParams } from '@/bindings/definitions';
>             ~~~~~~~~~~~~~~~~~~~~~~~
> 
> src/features/map_planning/store/MapStore.test.ts:6:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'PlantingDto'.
> 
> 6 import { PlantingDto } from '@/bindings/definitions';
>            ~~~~~~~~~~~
> 
> src/features/map_planning/store/MapStoreTypes.ts:1:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'PlantingDto'.
> 
> 1 import { PlantingDto } from '@/bindings/definitions';
>            ~~~~~~~~~~~
> 
> src/features/map_planning/store/RemoteActions.ts:9:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'Action'.
> 
> 9 import { Action as RemoteAction } from '@/bindings/definitions';
>            ~~~~~~
> 
> src/features/map_planning/store/TrackedMapStore.ts:9:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'PlantingDto'.
> 
> 9 import { PlantingDto } from '@/bindings/definitions';
>            ~~~~~~~~~~~
> 
> src/features/maps/api/createMap.ts:1:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'MapDto'.
> 
> 1 import { MapDto, NewMapDto } from '@/bindings/definitions';
>            ~~~~~~
> 
> src/features/maps/api/createMap.ts:1:18 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'NewMapDto'.
> 
> 1 import { MapDto, NewMapDto } from '@/bindings/definitions';
>                    ~~~~~~~~~
> 
> src/features/maps/api/findAllMaps.ts:1:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'MapDto'.
> 
> 1 import { MapDto, Page } from '@/bindings/definitions';
>            ~~~~~~
> 
> src/features/maps/components/MapCard.tsx:2:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'MapDto'.
> 
> 2 import { MapDto } from '@/bindings/definitions';
>            ~~~~~~
> 
> src/features/maps/components/MapCreateModal.tsx:1:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'NewMapDto'.
> 
> 1 import { NewMapDto } from '@/bindings/definitions';
>            ~~~~~~~~~
> 
> src/features/maps/routes/MapOverview.tsx:5:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'NewMapDto'.
> 
> 5 import { NewMapDto } from '@/bindings/definitions';
>            ~~~~~~~~~
> 
> src/features/seeds/components/CreateSeedForm.tsx:89:15 - error TS2322: Type '(search: string, options: SelectOption, additional: PageAdditionalInfo | undefined) => Promise<Page>' is not assignable to type 'LoadOptions<SelectOption, GroupBase<SelectOption>, PageAdditionalInfo>'.
>   Types of parameters 'options' and 'options' are incompatible.
>     Type 'readonly (SelectOption | GroupBase<SelectOption>)[]' is missing the following properties from type 'SelectOption': value, label
> 
> 89               loadOptions={loadPlants}
>                  ~~~~~~~~~~~
> 
>   src/components/Form/PaginatedSelectMenu.tsx:45:3
>     45   loadOptions: LoadOptions<Option, GroupBase<Option>, PageAdditionalInfo>;
>          ~~~~~~~~~~~
>     The expected type comes from property 'loadOptions' which is declared here on type 'IntrinsicAttributes & PaginatedSelectMenuProps<NewSeedDto, SelectOption, false>'
> 
> src/features/seeds/routes/CreateSeed.tsx:75:10 - error TS2741: Property 'onCancel' is missing in type '{ title: string; body: string; show: boolean; setShow: (showErrorModal: boolean) => void; submitBtnTitle: string; onSubmit: () => void; }' but required in type 'SimpleModalProps'.
> 
> 75         <SimpleModal
>             ~~~~~~~~~~~
> 
>   src/components/Modals/SimpleModal.tsx:15:3
>     15   onCancel: () => void;
>          ~~~~~~~~
>     'onCancel' is declared here.
> 
> src/features/seeds/routes/SeedDetails.tsx:91:10 - error TS2741: Property 'onCancel' is missing in type '{ title: string; body: string; show: boolean; setShow: Dispatch<SetStateAction<boolean>>; submitBtnTitle: string; onSubmit: () => void; }' but required in type 'SimpleModalProps'.
> 
> 91         <SimpleModal
>             ~~~~~~~~~~~
> 
>   src/components/Modals/SimpleModal.tsx:15:3
>     15   onCancel: () => void;
>          ~~~~~~~~
>     'onCancel' is declared here.
> 
> src/features/seeds/routes/ViewSeeds.tsx:68:10 - error TS2741: Property 'onCancel' is missing in type '{ title: string; body: string; show: boolean; setShow: Dispatch<SetStateAction<boolean>>; submitBtnTitle: string; onSubmit: () => void; }' but required in type 'SimpleModalProps'.
> 
> 68         <SimpleModal
>             ~~~~~~~~~~~
> 
>   src/components/Modals/SimpleModal.tsx:15:3
>     15   onCancel: () => void;
>          ~~~~~~~~
>     'onCancel' is declared here.
> 
> src/utils/enum.ts:1:10 - error TS2614: Module '"@/components/Form/SelectMenu"' has no exported member 'SelectOption'. Did you mean to use 'import SelectOption from "@/components/Form/SelectMenu"' instead?
> 
> 1 import { SelectOption } from '@/components/Form/SelectMenu';
>            ~~~~~~~~~~~~
> ```

A lot of these issues stem from the current master and appeared after I rebased.
I don't know how these made it into the master branch but I will try to fix them.

**Comment by absurd-turtle** [comment]
jenkins build please

File: frontend/src/components/Form/CreatableSelectMenu.stories.tsx
Comment (absurd-turtle) [reply frontend/src/components/Form/CreatableSelectMenu.stories.tsx:1]: Should be fixed now

**Comment by markus2330** [comment]
Still get plenty of errors (different ones):

```
> permaplant@0.1.3 doc
> typedoc src

[info] Loaded plugin typedoc-plugin-markdown
src/features/auth/api/getAuthInfo.ts:1:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'ConfigDto'.

1 import { ConfigDto } from '@/bindings/definitions';
           ~~~~~~~~~

src/features/map_planning/api/createPlanting.ts:1:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'NewPlantingDto'.

1 import { NewPlantingDto, PlantingDto } from '@/bindings/definitions';
           ~~~~~~~~~~~~~~

src/features/map_planning/api/createPlanting.ts:1:26 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'PlantingDto'.

1 import { NewPlantingDto, PlantingDto } from '@/bindings/definitions';
                           ~~~~~~~~~~~

src/features/map_planning/api/getPlantings.ts:1:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'PlantingDto'.

1 import { PlantingDto } from '@/bindings/definitions';
           ~~~~~~~~~~~

src/features/map_planning/api/movePlanting.ts:1:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'PlantingDto'.

1 import { PlantingDto, UpdatePlantingDto } from '@/bindings/definitions';
           ~~~~~~~~~~~

src/features/map_planning/api/movePlanting.ts:1:23 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'UpdatePlantingDto'.

1 import { PlantingDto, UpdatePlantingDto } from '@/bindings/definitions';
                        ~~~~~~~~~~~~~~~~~

src/features/map_planning/api/transformPlanting.ts:1:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'PlantingDto'.

1 import { PlantingDto, UpdatePlantingDto } from '@/bindings/definitions';
           ~~~~~~~~~~~

src/features/map_planning/api/transformPlanting.ts:1:23 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'UpdatePlantingDto'.

1 import { PlantingDto, UpdatePlantingDto } from '@/bindings/definitions';
                        ~~~~~~~~~~~~~~~~~

src/features/map_planning/layers/plant/actions.ts:9:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'PlantingDto'.

9 import { PlantingDto } from '@/bindings/definitions';
           ~~~~~~~~~~~

src/features/map_planning/routes/Map.tsx:10:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'ConnectToMapQueryParams'.

10 import { ConnectToMapQueryParams } from '@/bindings/definitions';
            ~~~~~~~~~~~~~~~~~~~~~~~

src/features/map_planning/store/MapStore.test.ts:6:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'PlantingDto'.

6 import { PlantingDto } from '@/bindings/definitions';
           ~~~~~~~~~~~

src/features/map_planning/store/MapStoreTypes.ts:1:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'PlantingDto'.

1 import { PlantingDto } from '@/bindings/definitions';
           ~~~~~~~~~~~

src/features/map_planning/store/RemoteActions.ts:9:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'Action'.

9 import { Action as RemoteAction } from '@/bindings/definitions';
           ~~~~~~

src/features/map_planning/store/TrackedMapStore.ts:9:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'PlantingDto'.

9 import { PlantingDto } from '@/bindings/definitions';
           ~~~~~~~~~~~

src/features/maps/api/createMap.ts:1:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'MapDto'.

1 import { MapDto, NewMapDto } from '@/bindings/definitions';
           ~~~~~~

src/features/maps/api/createMap.ts:1:18 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'NewMapDto'.

1 import { MapDto, NewMapDto } from '@/bindings/definitions';
                   ~~~~~~~~~

src/features/maps/api/findAllMaps.ts:1:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'MapDto'.

1 import { MapDto, Page } from '@/bindings/definitions';
           ~~~~~~

src/features/maps/components/MapCard.tsx:2:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'MapDto'.

2 import { MapDto } from '@/bindings/definitions';
           ~~~~~~

src/features/maps/components/MapCreateModal.tsx:1:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'NewMapDto'.

1 import { NewMapDto } from '@/bindings/definitions';
           ~~~~~~~~~

src/features/maps/routes/MapOverview.tsx:5:10 - error TS2305: Module '"@/bindings/definitions"' has no exported member 'NewMapDto'.

5 import { NewMapDto } from '@/bindings/definitions';
           ~~~~~~~~~
```

**Comment by markus2330** [comment]
I think we definitely should run `npm run doc` within the CI, can you add this to this PR?

**Comment by Bushuo** [comment]
@markus2330 
Those error seem like they only occur locally. Did you run `npm run generate-api-types`?

**Review by markus2330 — APPROVED** [review]
@Bushuo thx, this was it.

Still get a few warnings:

```
[warning] PathRouteCustomProps, defined in ./src/routes/types.ts, is referenced by routes/types.Routes but not included in the documentation.
[warning] DarkModeState, defined in ./src/features/dark_mode/store/DarkModeStore.ts, is referenced by features/dark_mode/store/DarkModeStore.useDarkModeStore.useDarkModeStore but not included in the documentation.
[warning] CreateSeedState, defined in ./src/features/seeds/store/CreateSeedStore.ts, is referenced by features/seeds/store/CreateSeedStore.default.default but not included in the documentation.
[warning] ProviderProps, defined in ./src/Providers.tsx, is referenced by Providers.Providers.Providers.__namedParameters but not included in the documentation.
[warning] PlantsLayerProps, defined in ./src/features/map_planning/layers/plant/PlantsLayer.tsx, is referenced by features/map_planning/layers/plant/PlantsLayer.default.default.__namedParameters but not included in the documentation.
```

and for `npm run storybook`:

```
WARN unable to find package.json for react-leaflet


WARN 🚨 Unable to index ./src/generated/docs/docs/modules/features_map_planning_store_MapStoreTypes.mdx:
WARN   Error: Unexpected character `` ` `` (U+0060) before name, expected a character that can start a name, such as a letter, `$`, or `_`
WARN     at /home/markus/Projekte/PermaplanT/frontend/node_modules/@storybook/core-server/dist/index.js:13:2054
WARN     at async Promise.all (index 82)
WARN     at async Promise.all (index 0)
WARN     at async StoryIndexGenerator.updateExtracted (/home/markus/Projekte/PermaplanT/frontend/node_modules/@storybook/core-server/dist/index.js:13:1601)
WARN     at async StoryIndexGenerator.ensureExtracted (/home/markus/Projekte/PermaplanT/frontend/node_modules/@storybook/core-server/dist/index.js:13:2357)
WARN     at async StoryIndexGenerator.initialize (/home/markus/Projekte/PermaplanT/frontend/node_modules/@storybook/core-server/dist/index.js:13:1528)
```

**Comment by Bushuo** [comment]
> and for npm run storybook:
> ...
> Error: Unexpected character `` ` `` (U+0060) before name, expected a character that can start a name
>
Ah, this is something that my plugin missed providing support for. I will add a fix for this.

**Comment by markus2330** [comment]
@Bushuo is this something introduced in this PR? (Please merge if the PR does not introduce the problem.)

Is there some way to see if `npm run storybook` would work for CI (without browser)? If you have an idea, please add to #438.

**Comment by Bushuo** [comment]
> @Bushuo is this something introduced in this PR? (Please merge if the PR does not introduce the problem.)
> 
> Is there some way to see if `npm run storybook` would work for CI (without browser)? If you have an idea, please add to #438.

No, this is not an issue of this PR.
I will take a look for storybook.

**Pull request merged**